contracts/modules/governance/MemberRoles.sol

```diff
@@ -280,43 +280,55 @@ contract MemberRoles is Governed, Iupgradable {
   }
 
   /**
-   * @dev Called by existed member if wish to switch membership to other address.
-   * @param _add address of user to forward membership.
+   * @dev switches membership for msg.sender to the specified address.
+   * @param newAddress address of user to forward membership.
    */
-  function switchMembership(address _add) external {
+  function switchMembership(address newAddress) external {
+    _switchMembership(msg.sender, newAddress);
+    tk.transferFrom(msg.sender, newAddress, tk.balanceOf(msg.sender));
+  }
 
-    require(!ms.isPause() && ms.isMember(msg.sender) && !ms.isMember(_add));
-    require(dAppToken.totalLockedBalance(msg.sender, now) == 0); // solhint-disable-line
-    require(!tf.isLockedForMemberVote(msg.sender)); // No locked tokens for Member/Governance voting
-    require(cr.getAllPendingRewardOfUser(msg.sender) == 0); // No pending reward to be claimed(claim assesment).
-    require(dAppToken.tokensUnlockable(msg.sender, "CLA") == 0, "Member should have no CLA unlockable tokens");
+  function switchMembershipForMember(address member, address newAddress) public onlyInternal {
+    _switchMembership(member, newAddress);
+  }
 
-    gv.removeDelegation(msg.sender);
-    dAppToken.addToWhitelist(_add);
-    _updateRole(_add, uint(Role.Member), true);
-    tk.transferFrom(msg.sender, _add, tk.balanceOf(msg.sender));
-    _updateRole(msg.sender, uint(Role.Member), false);
-    dAppToken.removeFromWhitelist(msg.sender);
+  /**
+   * @dev switches membership for member to the specified address.
+   * @param newAddress address of user to forward membership.
+   */
+  function _switchMembership(address member, address newAddress) internal {
+
+    require(!ms.isPause() && ms.isMember(member) && !ms.isMember(newAddress));
+    require(dAppToken.totalLockedBalance(member, now) == 0); // solhint-disable-line
```

Comment:
It is possible that the user has coverNotes that he didn't withdraw but they are already expired (out of grace period).